### PR TITLE
hapi(17.6): update docs, also update joi ecosys

### PR DESCRIPTION
**Pull request by SimonSchick**
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://hapijs.com/updates https://github.com/hapijs/joi/releases
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Also enabled a _few_ lint rules for joi.

### CODE REVIEW

**Comment by SimonSchick** [comment]
Only updating joi in the same go as hapi and joi are tightly coupled.

**Review by jhsimms — APPROVED** [review]

File: types/hapi/index.d.ts [hunk 1]
```diff
@@ -1531,6 +1545,12 @@ export interface RouteOptionsSecureObject {
      * boolean controlling the 'X-Content-Type-Options' header. Defaults to true setting the header to its only and default option, 'nosniff'.
      */
     noSniff?: boolean;
+
+    /**
+     * Controls the `Referrer-Policy` header, which has the following possible values.
+     * @default false Header will not be send.
+     */
+    referer?: false | RefererPolicy;
```
Comment (amcasey) [inline types/hapi/index.d.ts]: It looks like the library might spell it "referrer": https://github.com/hapijs/hapi/blob/76fcd7fa97747c92501b912d64db459d7172cb26/lib/config.js#L196

**Review by amcasey — CHANGES_REQUESTED** [review]
Possible typo.

File: types/hapi/index.d.ts (hunk 1 above)
Comment (SimonSchick) [reply types/hapi/index.d.ts]: 😱

**Comment by SimonSchick** [comment]
@amcasey ty, good catch!

**Comment by amcasey** [comment]
@SimonSchick looks like you missed one in the test. 😄

**Comment by SimonSchick** [comment]
@amcasey I even used the `rename` function, guess that doesn't work reliably 😛

**Review by amcasey — APPROVED** [review]

**Pull request merged**